rasa/nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -624,9 +624,9 @@ def predict_label(self, message):
             if X.any() and label_ids.size > 0:
 
                 # normalise scores if turned on
-                if self.loss_type == "softmax" and self.normalize_top_num_intents > 0:
-                    label_ids = label_ids[: self.normalize_top_num_intents]
-                    message_sim = message_sim[: self.normalize_top_num_intents]
+                if self.loss_type == "softmax" and self.ranking_length > 0:
+                    label_ids = label_ids[: self.ranking_length]
+                    message_sim = message_sim[: self.ranking_length]
                     message_sim = message_sim / np.sum(message_sim)
                 else:
                     label_ids = label_ids[:DEFAULT_LABEL_RANKING_LENGTH]
```

Comment:
I guess not in case of softmax, we still want to output `ranking_length` intents

Reply:
@Ghostvv I think we should either do:
1. ranking length is always used to be the output length as well as normalization, therefore if you set it to 50 for normalization purposes you should output 50. in this case if not softmax, it would also output 50 
2. ranking length is only used for normalization purposes, i.e. it is only a relevant parameter if the loss_type is softmax (we could make a user warning to that effect). the `DEFAULT_LABEL_RANKING_LENGTH` (could rename to `MAX_LABEL_RANKING_LENGTH`) is used for truncating output. In that case it would normalize over 50 and output 10. if not softmax, it would output 10. 

I think mixing it gets confusing, because the parameter has 2 jobs. What do you think, can you think of a solution that makes sense? maybe we should just make the actual output ranking_length configurable in addition to configuring the number of top confidences to normalize

Reply:
I agree with your point that it is confusing, I don't think it makes sense to output 10 if ranking length is 5, because the rest would be 5, also I'm not sure why would people want to have 50 there. I agree, with your point that it should be either of this two, I tend to the first case, but don't have strong opinion in there. Having two parameters to configure is I think unnecessary

Reply:
ok, i think basically what i had before (actually still?) was the first options, but also cutting off if normalization length is less than 10, so i think we agree

Reply:
Decision: `ranking_length` is used for normalization, and if you provide a lower ranking_length than 10 or number of intents, it will truncate to only the `ranking_length`. 

Otherwise 10 will be printed (for margin loss type as well)